### DISCUSSION TITLE
How we suppose to define of  the boundingbox of a dimension entity

### DISCUSSION THREAD

**Discussion by chibai** [mozman/ezdxf#644] [General]
for exmple in this dxf, when I calculate the boundingbox
[dimension_boundingbox.zip](https://github.com/mozman/ezdxf/files/8166049/dimension_boundingbox.zip)
```
from ezdxf import recover
from ezdxf import bbox

dxf_doc, dxf_auditor = recover.readfile('test.dxf')
msp = dxf_doc.modelspace()
cache = bbox.Cache()
# get overall bounding box
first_bbox = bbox.extents(msp, cache=cache)
print('width:', first_bbox.extmax[0] - first_bbox.extmin[0])
print('height:', first_bbox.extmax[1] - first_bbox.extmin[1])
print(first_bbox)
```
results is below
```
width: 2856316.0548670804
height: 170496.5272730448
[(-2822941.175259418, -154577.9702716459, 0.0), (33374.87960766246, 15918.55700139889, 0.0008776636679296)]
```

![9FBA4E1A-3EA2-4f25-97F0-AE5443B8BCBA](https://user-images.githubusercontent.com/5702248/156281920-a06c042f-c3aa-438b-932a-35817b825863.png)
Obviously, it consider the very far away single control vertex.
Intuitively for me, this is not what I suppose the dimension to be.
Of cause, it depends how EZDXF defines a boundingbox of dimension entity!!
I was wondering how can I get the visual part of boundingbox of a dimension?
Thank you~

**Comment by mozman** [comment]
All entities are decomposed into DXF primitives for the bounding box calculation. The visual representation of the DIMENSION entity is stored in an anonymous BLOCK which also contains a POINT entity for each definition point of the DIMENSION entity.
These POINTS are part of the bounding box and this is also what AutoCAD and BricsCAD do, result after "zoom to extends":

![image](https://user-images.githubusercontent.com/4230966/156294005-c3019171-92ac-4dda-8a5c-eaa917763474.png)

You can delete all POINT entities from the anonymous BLOCKS before the bounding box calculation, but don't save this DXF, because some 3rd party libraries may rely on the existence of these POINTS:

```Python
for block in dxf_doc.blocks:
    # all anonymous DIMENSION blocks start with "*D"
    if not block.name.startswith("*D"):
        continue
    for point in block.query("POINT"):
        point.destroy()
```

AutoCAD and BricsCAD **do not** complain about missing defpoints:

![image](https://user-images.githubusercontent.com/4230966/156294911-7fa3c1e2-edbe-4ff2-a3a8-734cd14c8041.png)
